### DISCUSSION TITLE
How to compile tree-sitter as dynamic library

### DISCUSSION THREAD

**Discussion by HerringtonDarkholme** [ast-grep/ast-grep#351] [General]
We don't have tree-sitter dynamic library instruction on the internet. But looking through the source code gives us some idea.

https://github.com/tree-sitter/tree-sitter/blob/a62bac5370dc5c76c75935834ef083457a6dd0e1/cli/loader/src/lib.rs#L111
```
export TREE_SITTER_LIBDIR=path/to/your/dir 
```
and then
```
tree-sitter test
```
will give you a dynamic lib at the `TREE_SITTER_LIBDIR`.

Alternatively, you can use these commands https://github.com/tree-sitter/tree-sitter/blob/a62bac5370dc5c76c75935834ef083457a6dd0e1/cli/loader/src/lib.rs#L380-L410

```
gcc -shared -fPIC -fno-exceptions -g -I header_path -o lib_path -O2
```

**Comment by ErickBessa** [comment]
This worked for me ! Thank so much!
